fenix/app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt

```diff
@@ -67,21 +68,31 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
         }
 
         override fun onLoggedOut() {
-            viewLifecycleOwner.lifecycleScope.launch {
-                findNavController().popBackStack()
+            forceCloseAndClearData()
+        }
+    }
 
-                // Remove the device name when we log out.
-                context?.let {
-                    val deviceNameKey = it.getPreferenceKey(R.string.pref_key_sync_device_name)
-                    preferenceManager.sharedPreferences?.edit()?.remove(deviceNameKey)?.apply()
-                }
+    private fun forceCloseAndClearData() {
```

Comment:
nit: naming. Force close on Android has a different meaning (kill the app).

```suggestion
    private fun closeAndClearData() {
```